### ISSUE TITLE
Invalid code gen when assigning items to a collection property in XAML

### ISSUE THREAD

**Issue by Daoting** [unoplatform/uno#211]
## I'm submitting a...

 - Bug report [#189](189)    The difference is that it is the root element.
 

## Current behavior

xaml like this:
```
<local:SingleWin>
	<local:SingleWin.Bar>
		<local:Btn />
		<local:Btn />
		<local:Btn />
	</local:SingleWin.Bar>
</local:SingleWin>
```
local:SingleWin is root element. local:SingleWin.Bar is collection instance.

The Bar's define:
`public IList<IBarElement> Bar { get { return _bar.Commands; } }`

XamlFileGenerator code:
```
this
.Apply((c9 =>
{
// Source Frame\PageWinDemo.xaml (Line 1:2)
// WARNING Property c9.base does not exist on {using:Dt.Base}PageWin, the namespace is http://www.w3.org/XML/1998/namespace. This error was considered irrelevant by the XamlFileGenerator
c9.Bar = 
{
	new global::Dt.Base.Btn
	{
		// Source ..\..\..\..\..\..\Frame\PageWinDemo.xaml (Line 8:10)
	}
	,
	new global::Dt.Base.Btn
	{
		// Source ..\..\..\..\..\..\Frame\PageWinDemo.xaml (Line 9:10)
	}
	,
	new global::Dt.Base.ToggleBtn
	{
		// Source ..\..\..\..\..\..\Frame\PageWinDemo.xaml (Line 20:10)
	}
	,
}
;

```
## Environment

```
Nuget Package: 

Package Version(s): 

Affected platform(s):
- [x] iOS
- [x] Android
- [x] WebAssembly
- [ ] Windows
- [ ] Build tasks

Visual Studio
- [x] 2017 (version: )
- [ ] 2017 Preview (version: )
- [ ] for Mac (version: )

Relevant plugins
- [ ] Resharper (version: )
```

**Comment by jeromelaban** [comment]
Agreed, the management of top level items is a bit different from the others, this will have to be adressed.

**Comment by carldebilly** [comment]
Fixed by PR #1356
